front-end/src/views/function/components/AddFunc.vue

```diff
@@ -0,0 +1,242 @@
+                    style="width: 66%"
+                    prop="className">
+        <el-input v-model="form.className"
+                  placeholder="请输入ClassName"
+                  clearable>
+        </el-input>
+      </el-form-item>
+      <el-form-item label="Runtime:"
+                    style="width: 30%"
+                    prop="runtime">
+        <el-select v-model="form.runtime">
+          <el-option label="JAVA"
+                     value="JAVA">
+          </el-option>
+          <el-option label="GO"
+                     value="GO">
+          </el-option>
+        </el-select>
+      </el-form-item>
+      <el-form-item class="upload"
+                    label="File:"
+                    prop="file"
+                    style="width: 100%">
+        <el-upload
+          :limit="1"
+          drag
+          ref="file"
+          accept=".jar"
+          action="0.0//不写会报错，神奇"
+          :auto-upload="false"
+          :on-change="getFile">
+          <i class="el-icon-upload"></i>
+          <div class="el-upload__text">将文件拖到此处，或点击上传</div>
+          <div class="el-upload__tip">目前只支持jar文件上传</div>
+        </el-upload>
+      </el-form-item>
+      <div
+        :style="{
+          position: 'absolute',
+          right: 0,
+          bottom: 0,
+          width: '100%',
+          borderTop: '1px solid #e9e9e9',
+          padding: '10px 16px',
+          background: '#fff',
+          textAlign: 'right',
+          zIndex: 1
+        }">
+        <el-button type="primary" @click="onSubmit">立即创建</el-button>
+        <el-button @click="resetField('addForm')">重置</el-button>
+      </div>
+    </el-form>
+  </el-drawer>
+</template>
+
+<script>
+import { create } from '@/api/func'
+export default {
+  props:{
+    functionList: {
+      type: Object,
+      default: () => {},
+    },
+    refresh: {
+      type: Function,
+      default: () => {},
+    }
+  },
+  data() {
+    return {
+      file:'',
+        form: {
+        functionName: '',
+        inputs: [''],
+        output: '',
+        logTopic: '',
+        className: '',
+        runtime: 'JAVA'
+      },
+      rules:{
+        functionName:[
+          { required: true, message: '请输入FunctionName', trigger: 'blur' }
+        ],
+        output:[
+          { required: true, message: '请输入Output', trigger: 'blur' }
+        ],
+        logTopic:[
+          { required: true, message: '请输入Log-topic', trigger: 'blur' }
+        ],
+        className:[
+          { required: true, message: '请输入ClassName', trigger: 'blur' }
+        ]
+      }
+    }
+  },
+  methods: {
+    onSubmit() {
+      this.$refs.addForm.validate(async valid =>{
+        if(!valid) return
+        if(!this.file){
+            this.$message('还没添加文件')
+          return
+        }
+        const functionConfig = JSON.parse(JSON.stringify(this.form).replace(/logTopic/g,'log-topic'))
+        const functionName = functionConfig.FunctionName
+        delete functionConfig.functionName
```

Comment:
Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/FunctionStream/function-stream/pullRequest?prid=7834759)
```suggestion
          delete functionConfig.functionName
```